commands/liquidity_provision_submission_test.go

```diff
@@ -182,3 +184,225 @@ func TestLiquidityProvisionSubmission(t *testing.T) {
+	type args struct {
+		cmd *commandspb.LiquidityProvisionAmendment
+	}
+	tests := []struct {
+		name      string
+		args      args
+		wantErr   assert.ErrorAssertionFunc
+		errString string
+	}{
+		{
+			name: "Should return an error when the command is nil",
+			args: args{
+				cmd: nil,
+			},
+			wantErr:   assert.Error,
+			errString: "liquidity_provision_amendment (is required)",
+		},
+		{
+			name: "Should return an error when id is not provided",
+			args: args{
+				cmd: &commandspb.LiquidityProvisionAmendment{
+					Id:       "",
+					MarketId: "abcd",
+					Sells: []*types.LiquidityOrder{
+						{Reference: types.PeggedReference_PEGGED_REFERENCE_BEST_ASK, Offset: 10, Proportion: 1},
+					},
+					Buys: []*types.LiquidityOrder{
+						{Reference: types.PeggedReference_PEGGED_REFERENCE_BEST_BID, Offset: -10, Proportion: 1},
+					},
+				},
+			},
+			wantErr:   assert.Error,
+			errString: "liquidity_provision_amendment.id (is required)",
+		},
+		{
+			name: "Should return an error when market_id is not provided",
+			args: args{
+				cmd: &commandspb.LiquidityProvisionAmendment{
+					Id:       "123",
+					MarketId: "",
+					Sells: []*types.LiquidityOrder{
+						{Reference: types.PeggedReference_PEGGED_REFERENCE_BEST_ASK, Offset: 10, Proportion: 1},
+					},
+					Buys: []*types.LiquidityOrder{
+						{Reference: types.PeggedReference_PEGGED_REFERENCE_BEST_BID, Offset: -10, Proportion: 1},
+					},
+				},
+			},
+			wantErr:   assert.Error,
+			errString: "liquidity_provision_amendment.market_id (is required)",
+		},
+		{
+			name: "Should return no errors if amendment changes nothing",
+			args: args{
+				cmd: &commandspb.LiquidityProvisionAmendment{
+					Id:       "123",
+					MarketId: "abcd",
+					Sells:    nil,
+					Buys:     nil,
+				},
+			},
+			wantErr: assert.NoError,
+		},
+		{
+			name: "Should return no errors if amendment buys and sells are balanced",
+			args: args{
+				cmd: &commandspb.LiquidityProvisionAmendment{
+					Id:       "123",
+					MarketId: "abcd",
+					Sells: []*types.LiquidityOrder{
+						{Reference: types.PeggedReference_PEGGED_REFERENCE_BEST_ASK, Offset: 10, Proportion: 1},
+					},
+					Buys: []*types.LiquidityOrder{
+						{Reference: types.PeggedReference_PEGGED_REFERENCE_BEST_BID, Offset: -10, Proportion: 1},
+					},
+				},
+			},
+			wantErr: assert.NoError,
+		},
+		{
+			name: "Should return an error if sell side shape is provided with no buy side shape",
+			args: args{
+				cmd: &commandspb.LiquidityProvisionAmendment{
+					Id:       "123",
+					MarketId: "abcd",
+					Sells: []*types.LiquidityOrder{
+						{Reference: types.PeggedReference_PEGGED_REFERENCE_BEST_ASK, Offset: 10, Proportion: 1},
+					},
+					Buys: []*types.LiquidityOrder{},
+				},
+			},
+			wantErr:   assert.Error,
+			errString: "liquidity_provision_submission.buys (empty shape)",
+		},
+		{
+			name: "Should return an error if buy side shape is provided with no sell side shape",
```

Comment:
This should not be an issue for an amendment? That would suggest that no amendment are done on the sell side and it stay as it was.

Reply:
Or maybe that differenciate between an empty slice and nil? Not sure that proto would do any differenciation at this point between a nil and empty slice.

Reply:
Yes, sorry, you are right. This validation is required on submission, but not on validation as an empty/nil slice means no changes to the existing submission. I will correct this.